### PR TITLE
Convert unknown implementation warning to error

### PR DESCRIPTION
**Pull request by wolfs**
This was deprecated for Gradle 8.0, so let's make it an error now.

### CODE REVIEW

**Review by nathan-contino — APPROVED** [review]
Docs LGTM.

**Review by lptr — APPROVED** [review]
LGTM, left a small comment.

File: subprojects/core/src/integTest/groovy/org/gradle/api/internal/tasks/SnapshotTaskInputsOperationIntegrationTest.groovy [hunk 1]
```diff
@@ -150,15 +149,7 @@ class SnapshotTaskInputsOperationIntegrationTest extends AbstractIntegrationSpec
...
-        expectImplementationUnknownDeprecation {
-            implementationOfTask(':customTask')
-            unknownClassloader('CustomTask_Decorated')
-        }
-        expectImplementationUnknownDeprecation {
-            additionalTaskAction(':customTask')
-            unknownClassloader('CustomTask_Decorated')
-        }
-        succeeds('customTask', '--build-cache')
+        fails('customTask', '--build-cache')
```
[the remark this answers is not in the document]
Comment (wolfs) [reply subprojects/core/src/integTest/groovy/org/gradle/api/internal/tasks/SnapshotTaskInputsOperationIntegrationTest.groovy:156]: Good point, done.

**Comment by wolfs** [comment]
@bot-gradle test and merge

File: subprojects/execution/src/main/java/org/gradle/internal/execution/steps/ValidateStep.java [hunk 2]
```diff
@@ -209,7 +209,7 @@ private <T extends ValidationProblemBuilder<T>> T configureImplementationValidat
         return problem
             .typeIsIrrelevantInErrorMessage()
             .withId(ValidationProblemId.UNKNOWN_IMPLEMENTATION)
-            .reportAs(Severity.WARNING)
+            .reportAs(Severity.ERROR)
```
[the remark this answers is not in the document]
Comment (wolfs) [reply subprojects/execution/src/main/java/org/gradle/internal/execution/steps/ValidateStep.java:212]: I created an issue for this: https://github.com/gradle/gradle/issues/21692

**Pull request merged**